core/src/test/scala/org/broadinstitute/dsde/rawls/user/UserServiceSpec.scala

```diff
@@ -1238,6 +1247,249 @@ class UserServiceSpec
     Await.result(userService.getBillingProject(projectName), Duration.Inf) shouldEqual expected
   }
 
+  it should "not attempt to update landing zones for non-azure projects" in {
+    val billingProfile = new ProfileModel().id(UUID.randomUUID()).cloudPlatform(BPMCloudPlatform.GCP)
+    val projectName = RawlsBillingProjectName(UUID.randomUUID().toString)
+    val project = RawlsBillingProject(
+      projectName,
+      CreationStatuses.CreatingLandingZone, // normally this is invalid, but it shouldn't matter because this isn't an azure project
+      None,
+      None,
+      billingProfileId = Some(billingProfile.getId.toString)
+    )
+    val userService = spy(getUserService())
+    val result = userService.mapCloudPlatform(project, Some(billingProfile), Set())
+    result shouldBe RawlsBillingProjectResponse(Set(), project, CloudPlatform.GCP)
+    verify(userService, never).updateLandingZoneStatus(ArgumentMatchers.any())
+  }
+
+  it should "not update the billing project for a landing zone if the project is not waiting for landing zone creation" in {
+    val billingProfile = new ProfileModel().id(UUID.randomUUID()).cloudPlatform(BPMCloudPlatform.AZURE)
+    val projectName = RawlsBillingProjectName(UUID.randomUUID().toString)
+    val project = RawlsBillingProject(
+      projectName,
+      CreationStatuses.Ready,
+      None,
+      None,
+      azureManagedAppCoordinates = Some(AzureManagedAppCoordinates(null, null, null)),
+      billingProfileId = Some(billingProfile.getId.toString)
+    )
+    val userService = spy(getUserService())
+    val result = userService.mapCloudPlatform(project, Some(billingProfile), Set())
+    result shouldBe RawlsBillingProjectResponse(Set(), project, CloudPlatform.AZURE)
+    verify(userService).updateLandingZoneStatus(project) // update landing status was called, but it was a no-op
```

Comment:
Is there a way to verify that it was a no-op?

Reply:
I'm using the default mocks, so it doing anything that's not a no-op will make it blow up